src/app/modules/shared-components/components/input/input.component.ts

```diff
@@ -7,23 +7,25 @@ import {
   OnChanges,
   SimpleChanges,
 } from '@angular/core';
+import { FormGroup } from '@angular/forms';
 
 @Component({
   selector: 'alg-input',
   templateUrl: './input.component.html',
   styleUrls: ['./input.component.scss'],
 })
 export class InputComponent implements OnInit, OnChanges {
-  @Input() value: string;
-  @Input() placeholder: string;
+  @Input() name : string | number | null; // name of the input in the parent form
+
+  @Input() placeholder = ''; // avoid 'undefined' if no placeholder specified
   @Input() icon: string;
   @Input() mode = 'dark';
   @Input() type = 'small';
   @Input() hasButton = false;
   @Input() inputType = 'text';
   @Input() leftIcon = 'fa fa-font';
```

Comment:
Actually, this input component, as part of the design is not really valid. As you are the first one to re-use it, it is your responsibility to fix it: (should have been done as an initial PR ideally)
(I haven't checked the html, it is just my feeling when reading this ts)

- `icon` is not optional but not defined in constructor. Probably as in score ring it should be `@Input() icon?: string; // a font-awesome icon identifier` 
- but what is `leftIcon` then? (maybe 'icon' is not clear enough). Having 'fa' specified here is inconsistent with score ring which supposed it was implicit.   
- mode, type, inputType, leftIcon are not just string, they have only some allowed values
- mode and type are quite very bad names which too general which match everything and nothing. If mode is dark and light, it could be a boolean... type may be "size" with only few possible values.
- hasButton, I don't get when reading it what it means, so it probably needs a short comment

Reply:
The 'leftIcon'is the icon of the Input, while the 'icon' is the icon of the button that can be added using the hasButton, I've renamed them for clarity. 

I'm more confused with inputType, leftIcon only having some allowed values. inputType is used as the type of the input which, according to typescript's prototype for input, is a string. As for leftIcon it's supposed to also be a Font Awesome icon which means a lot of different possibilities.

Reply:
- there is no need for hasButton if the the giving an button icon enables it.
- Input type are the html valid input. Ok we could say that's a string as I see there are many valid values.
- for leftIcon, it is a mistake, of course this is not a limited number of values

Reply:
done !